src/Extension/FluentExtension.php

```diff
@@ -48,6 +48,7 @@
  *
  * @template T of DataObject
  * @extends DataExtension<T&static>
+ * @property DataObject|$this $owner
```

Comment:
```suggestion
```
There are already generics set up for this, which is what the `@template` and `@extends` are for.